### PR TITLE
Keyword-style argument API examples

### PR DESCRIPTION
**Pull request by nathanielknight**
This PR contains:

- Some bug-fixes to the way Kive handles keyword-style pipeline parameters.
- API examples demonstrating how to start pipelines with keyword style parameters.

See individual commit messages for further explanations.

This PR pertains to #511

### CODE REVIEW

File: kive/container/management/commands/runcontainer.py [hunk 1]
```diff
@@ -86,7 +89,12 @@ def fill_sandbox(self, run):
         input_path = os.path.join(run.full_sandbox_path, 'input')
         os.mkdir(input_path)
         for dataset in run.datasets.all():
-            target_path = os.path.join(input_path, dataset.argument.name)
+            if dataset.argument.argtype in (
+                    ContainerArgumentType.OPTIONAL_MULTIPLE_INPUT,
+                    ContainerArgumentType.OPTIONAL_INPUT):
+                target_path = os.path.join(input_path, dataset.dataset.name)
```
Comment (donkirkby) [inline kive/container/management/commands/runcontainer.py:95]: I'm worried about name collisions here. Would it be safer to add a number to the argument name when needed? To be extra paranoid, keep a set of file names you've created so far, and check all arguments for collisions. That way `--option*` can't collide with `--option2` when you pass in a second file.

Comment (nathanielknight) [reply kive/container/management/commands/runcontainer.py:95]: If the filenames on disk aren't super important, could we just tack a statistically-unique random string to the end of each filename as we copy them in? That lets us handle each file without reference to the others.

Comment (donkirkby) [reply kive/container/management/commands/runcontainer.py:95]: OK, but we can do better than statistically unique. Each dataset has a unique id, so add that to the file name. Preferably, keep the file extension unchanged by adding the id between the base name and the extension.

File: api/example_5.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+"Run the example collation pipeline, which has multiple inputs."
```
Comment (donkirkby) [inline api/example_5.py:1]: This example is for keyword args, not multivalued inputs.

File: api/example_6.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+"Run the example collation pipeline, which has multiple inputs."
```
Comment (donkirkby) [inline api/example_6.py:1]: I think it's clearer to say "multivalued inputs" instead of "multiple inputs".

**Comment by nathanielknight** [comment]
@donkirkby comments have been addressed.

**Comment by nathanielknight** [comment]
Sorry, an auto-formatter got loose on one of the files 🤦‍♂️

**Pull request merged**